packages/adblocker/src/engine/engine.ts

```diff
@@ -776,11 +789,21 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
         } else if (filter.isGenericHide() || filter.isSpecificHide()) {
           hideExceptions.push(filter);
         } else if (filter.isException()) {
-          exceptions.push(filter);
+          if (filter.isRemoveParam()) {
+            if (this.config.enableHtmlFiltering) {
+              substitudes.push(filter);
+            }
+          } else {
+            exceptions.push(filter);
+          }
         } else if (filter.isImportant()) {
           importants.push(filter);
         } else if (filter.isRedirect()) {
           redirects.push(filter);
+        } else if (filter.isRemoveParam()) {
+          if (this.config.enableHtmlFiltering) {
```

Comment:
How are `enableHtmlFiltering` and `removeparam` related?

Reply:
`removeparam` requires html filtering capability even if they don't pass the html filtering code flow. Alternatively, I believe html filtering isn't the one I'd like to express but it's the closest existing in the library.

Reply:
I removed the flag requirement in 22cd3d71075ce5279916e6b88f56fbbde657beee but keeping in mind about having more variable.